app/jobs/education_form/create_daily_spool_files.rb

```diff
@@ -16,38 +18,58 @@ class CreateDailySpoolFiles < ActiveJob::Base
       retry_job(wait: 1.minute)
     end
 
-    def perform(day = Date.yesterday)
-      # TODO: add and use the mapping of schools/regions -> spool file
-
-      records = EducationBenefitsClaim.unprocessed_for(day)
+    def perform
+      # Fetch all the records for the day
+      records = EducationBenefitsClaim.unprocessed
+      # Group the formatted records into different regions
+      regional_data = group_submissions_by_region(records)
+      # Create a remote file for each region, and write the records into them
+      create_files(regional_data)
+      # mark the records as processed
+      records.update_all(processed_at: Time.zone.now)
+      # TODO: Log the success/failure of the submission somewhere
+      true
+    end
 
+    def group_submissions_by_region(records)
+      regional_data = Hash.new { |h, k| h[k] = [] }
       records.each do |record|
-        format_application(record)
-      end.join(WINDOWS_NOTEPAD_LINEBREAK)
-
-      Tempfile.create('education_spoolfile') do |f|
-        f.write dev_spool_output
-        # if you want to roughly(ish) see what the generated image looks like
-        # convert -pointsize 10 -density 200 -depth 8 -font Courier text:#{f.path} sample.win.tiff
+        form = record.open_struct_form
+        region_key = EducationFacility.region_for(form)
+        regional_data[region_key] << form
       end
+      regional_data
+    end
 
-      # TODO: SFTP the generated file(s)
-      # TODO: Mark the applications as processed once the send is successful
-      # TODO: Log the success/failure of the submission somewhere
-      true
+    def create_files(structured_data)
+      # TODO: Will be implemented in a follow-up PR.
+      Net::SFTP.start('host', 'username', password: 'password') do |sftp|
+        structured_data.each_with_object({}) do |(region, records), _localfiles|
```

Comment:
Will `_localfiles` ever be used? Looking at this, you could just iterate over the hash with `.each` and not `.each_with_object`